pattern.go

```diff
@@ -196,12 +197,33 @@ func readSpecialPattern(pb *patternBuild, valsIn []typedVal) (pathVals []typedVa
 		pathVals, err = readExistsSpecial(pb, pathVals)
 	case "shellstyle":
 		pathVals, err = readShellStyleSpecial(pb, pathVals)
+	case "prefix":
+		pathVals, err = readPrefixSpecial(pb, pathVals)
 	default:
 		err = errors.New("unrecognized in special pattern: " + tt)
 	}
 	return
 }
 
+func readPrefixSpecial(pb *patternBuild, valsIn []typedVal) (pathVals []typedVal, err error) {
+	t, err := pb.jd.Token()
+	if err != nil {
+		return
+	}
+	pathVals = valsIn
+
+	prefixString, ok := t.(string)
+	if !ok {
+		err = errors.New("value for 'prefix' must be a string")
+		return
+	}
+	pathVals = append(pathVals, typedVal{vType: prefixType, val: `"` + prefixString + `"`})
+
+	// has to be } or tokenizer will throw error
+	_, err = pb.jd.Token()
+	return
```

Comment:
In general I don't like named returns in Go functions (readability). Any reason we need named returns here?

Reply:
I confess that this is the one area where I disagree with the Go community; as a result, there are named returns (heh, "naked" returns) here and there around Quamina. The linter doesn't seem to care.

For the moment, I don't feel any urgency about chasing them out. If someone cared enough I'd approve a PR that takes them all out systematically.